client/src/components/HomePage/HomepageBody.jsx

```diff
@@ -27,7 +29,22 @@ const HomepageBody = () => {
         warn("test");
     }
 
+    const handleCloseDocumentOptionsModal = () => {
+        setIsDocumentOptionsModalOpen(false);
+    }
+
+    const handleDocumentOptionsClick = (e) => {
+        e.stopPropagation();
+        const buttonRect = e.currentTarget.getBoundingClientRect();
+        setModalPosition({
+            left: buttonRect.left,
+            bottom: buttonRect.bottom
+        });
```

Comment:
using `useState` for this might not be the best approach.  It will cause a re-render cycle before its updated and may cause the UI to jump around.  I think `useRef` is more appropriate here.